honeybee_schema/radiance/material.py

```diff
@@ -0,0 +1,280 @@
+                    '(default: 1.52).'
+    )
+
+
+class BSDF(ModifierBase):
+    """Radiance BSDF (Bidirectional Scattering Distribution Function) material."""
+
+    type: constr(regex='^BSDF$') = 'BSDF'
+
+    up_orientation: List[float] = Field(
+        default=(0.01, 0.01, 1.00),
+        description='Vector as sequence that sets the hemisphere that the BSDF material '
+                    'faces. (default: (0.01, 0.01, 1.00).'
+    )
+
+    thickness: float = Field(
+        default=0,
+        description='Optional number to set the thickness of the BSDF material '
+                    'Sign of thickness indicates whether proxied geometry is '
+                    'behind the BSDF surface (when thickness is positive) or in '
+                    'front (when thickness is negative)(default: 0).'
+    )
+
+    function_file: str = Field(
+        default='.',
+        min_length=1,
+        max_length=100,
+        description='Optional input for function file (default: ".").'
+    )
+
+    transform: str = Field(
+        default=None,
+        min_length=1,
+        max_length=100,
+        description='Optional transform input to scale the thickness and reorient '
+                    'the up vector (default: None).'
+    )
+
+    bsdf_data: bytes = Field(
+        ...,
+        description='BSDF xml file data as bytes.'
+    )
+
+    front_diffuse_reflectance: List[float] = Field(
+        default=None,
+        description='Optional additional front diffuse reflectance as sequence of '
+                    'numbers (default: None).'
+    )
+
+    back_diffuse_reflectance: List[float] = Field(
+        default=None,
+        description='Optional additional back diffuse reflectance as sequence of '
+                    'numbers (default: None).'
+    )
+
+    diffuse_transmittance: List[float] = Field(
+        default=None,
+        description='Optional additional diffuse transmittance as sequence of '
+                    'numbers (default: None).'
+    )
+
+
+class Glow(ModifierBase):
+    """Radiance Glow material."""
+
+    type: constr(regex='^Glow$') = 'Glow'
+
+    r_emittance: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the red channel glow '
+                    '(default: 0).'
+    )
+
+    g_emittance: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the green channel glow '
+                    '(default: 0).'
+    )
+
+    b_emittance: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the blue channel glow '
+                    '(default: 0).'
+    )
+
+    max_radius: float = Field(
+        default=0,
+        description='Maximum radius for shadow testing (default: 0). '
+    )
+
+
+class Light(ModifierBase):
+    """Radiance Light material."""
```

Comment:
Maybe you could just have `Glow` inherit from `Light` since they share 3 of the same properties.

Reply:
Yeah I agree. The only reason I kept it was because the docstrings for both mentioned 'glow' vs 'light', but on second thought this works better if we abstract it.